x-pack/plugins/infra/public/components/log_stream/log_stream.stories.mdx

```diff
@@ -394,6 +394,38 @@ The rendering of the column headers and the cell contents can also be customized
   </Story>
 </Canvas>
 
+### Render props for overwritting states
+
+By default the `<LogStream />` component will provide default empty and loading states. But if you were to need your own you can use the following props in order to overwrite it:
+
+```tsx
+<LogStream
+  sourceId="my_source"
+  startTimestamp={...}
+  endTimestamp={...}
+  loadingState={<p>Loading...</p>}
+  emptyState={({ handleReload }) => (
+    <div>
+      <p>No entries.</p>
+      <button>Click here to reload</button>
+    </div>
+  )}
+/>
+```
+
+### Hidding column headers
+
+By default the `<LogStream />` component will always show the column headers. We can control that by using the `hasColumnHeaders` prop:
+
+```tsx
+<LogStream
+  sourceId="my_source"
+  startTimestamp={...}
+  endTimestamp={...}
+  hasColumnHeaders={false}
+/>
+```
+
```

Comment:
How about rendering those as actual stories?

Reply:
I added the stories for rendering those 2 new sections, though I couldn't find a way to show the empty state since there doesn't seem to be a way to overwrite the dataMock passed to the context provider.